rslint_core/src/groups/errors/valid_typeof.rs

```diff
@@ -0,0 +1,116 @@
+use crate::rule_prelude::*;
+use ast::{BinExpr, Expr, UnaryOp};
+use rslint_parser::{SyntaxText, TextRange};
+use SyntaxKind::*;
+
+declare_lint! {
+    /**
+    Enforces the to use valid string literals in a `typeof` comparison.
+
+    ## Invalid Code Examples
+    ```ignore
+    typeof foo === "strnig"
+    typeof foo == "undefimed"
+    typeof bar != "nunber"
+    typeof bar !== "fucntion"
+    ```
+    */
+    #[derive(Default)]
+    ValidTypeof,
+    errors,
+    "valid-typeof"
+}
+
+const VALID_TYPES: [&str; 8] = [
+    "undefined",
+    "object",
+    "boolean",
+    "number",
+    "string",
+    "function",
+    "symbol",
+    "bigint",
+];
+
+#[typetag::serde]
+impl CstRule for ValidTypeof {
+    fn check_node(&self, node: &SyntaxNode, ctx: &mut RuleCtx) -> Option<()> {
+        if node.kind() == BIN_EXPR {
+            let expr = node.to::<BinExpr>();
+            if !expr.comparison() {
+                return None;
+            }
+
+            let (literal, range) = get_type_literal(expr.lhs(), expr.rhs())?;
+
+            if !VALID_TYPES.iter().any(|ty| *ty == literal) {
+                let literal = String::from(literal);
+                let suggestion = VALID_TYPES
+                    .iter()
+                    .map(|ty| (*ty, strsim::levenshtein(ty, &literal)))
+                    .filter(|(_, d)| *d < 4)
+                    .next();
+
+                let err = ctx.err(self.name(), "Invalid typeof comparison value");
```

Comment:
```suggestion
                let err = ctx.err(self.name(), "invalid typeof comparison value");
```
error messages should be lowercase, same with labels, i know this isnt the case in all of the linter but i would like to enforce it from now on and eventually go back and fix the rest.